presto-native-execution/presto_cpp/main/TaskManager.cpp

```diff
@@ -173,6 +174,78 @@ std::unique_ptr<TaskInfo> TaskManager::createOrUpdateErrorTask(
   return std::make_unique<TaskInfo>(info);
 }
 
+// If the plan fragment contains operator that can spill and spilling is
+// enabled, created the spilling directory for the task.
+static std::string prepareTaskSpillDirectory(
+    const velox::core::QueryCtx* queryCtx,
+    const TaskId& taskId,
+    const velox::core::PlanFragment& planFragment) {
+  if (not queryCtx->queryConfig().spillEnabled()) {
+    return "";
+  }
+
+  const auto baseSpillPath = SystemConfig::instance()->spillerSpillPath();
+  if (baseSpillPath.empty()) {
+    return "";
+  }
+
+  // Functors to check if a given node can spill.
+  const auto hasSpillingHashAggrFunc = [](const core::PlanNode* node) {
+    if (node->name() == "Aggregation") {
+      if (const auto* aggregationNode =
+              dynamic_cast<const core::AggregationNode*>(node)) {
+        if (aggregationNode->isFinal() || aggregationNode->isSingle()) {
+          return true;
+        }
+      }
+    }
+    return false;
+  };
+  const auto hasSpillingOrderByFunc = [](const core::PlanNode* node) {
+    return (node->name() == "OrderBy");
+  };
+  const auto hasSpillingHashJoinFunc = [](const core::PlanNode* node) {
+    return (node->name() == "HashJoin");
+  };
+
+  // Determine if this task can spill at all, to avoid creating a folder for
+  // nothing. Looks for particular nodes for that.
+  const auto* topNode = planFragment.planNode.get();
+  if ((queryCtx->queryConfig().aggregationSpillEnabled() &&
+       core::PlanNode::findFirstNode(topNode, hasSpillingHashAggrFunc)) ||
+      (queryCtx->queryConfig().orderBySpillEnabled() &&
+       core::PlanNode::findFirstNode(topNode, hasSpillingOrderByFunc)) ||
+      (queryCtx->queryConfig().joinSpillEnabled() &&
+       core::PlanNode::findFirstNode(topNode, hasSpillingHashJoinFunc))) {
+    // Might need spilling.
+  } else {
+    return "";
+  }
+
+  // Form the task spill path
+  const std::string& queryId = queryCtx->queryId();
+  std::stringstream ss;
+  ss << baseSpillPath << "/";
+  // Generate 'YYYY-MM-DD' from the query ID, which starts with 'YYYYMMDD'.
+  ss << fmt::format(
+      "{}-{}-{}",
+      queryId.substr(0, 4),
+      queryId.substr(4, 2),
+      queryId.substr(6, 2));
+  ss << "/";
+  // TODO(spershin): We will like need to use identity (from config?) in the
+  // long run. Use 'presto_native' for now.
+  ss << "presto_native"
+     << "/spill/" << queryId << "/" << taskId << "/";
```

Comment:
I guess we don't need spill path and I assume the base directory configured in system config should be dedicated used for spill?

Reply:
The answer is yes, I believe, although I don't fully understand the question.

Reply:
I mean we might not need "/spill/" in the path.

Reply:
Hmm, I don't have a string opinion on that, I thought we might write  some other data, not just spill.
I will remove it.